### PR TITLE
Add the Authorization vocabularies annotation into core edm model

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

* No related issue created.

### Description

OData to Open API will use the "Authorization" vocabularies to annotate the request/response.
Meantime, OASIS approve the "Authorization" vocabularies, see the definition at: https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Authorization.V1.xml

This PR is to add the Authorization vocabularies into the EDM code lib, then the customer can use the term defined in Authorization vocabularies to add annotation.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by xuzhg** [comment]
@mikepizzo @robward-ms @AlanWong-MS Would you please take some time now on this review? Thanks a lot.

**Comment by AlanWong-MS** [comment]
typo in title: Authoritation -> Authorization

File: src/Microsoft.OData.Edm/GlobalSuppressions.cs [hunk 1]
```diff
@@ -243,6 +243,7 @@
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Scope = "member", Target = "Microsoft.OData.Edm.Vocabularies.V1.CapabilitiesVocabularyModel.#.cctor()")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Scope = "member", Target = "Microsoft.OData.Edm.Vocabularies.V1.CoreVocabularyModel.#.cctor()")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Scope = "member", Target = "Microsoft.OData.Edm.Vocabularies.Community.V1.AlternateKeysVocabularyModel.#.cctor()")]
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Scope = "member", Target = "Microsoft.OData.Edm.Vocabularies.Community.V1.AuthorizaationVocabularyModel.#.cctor()")]
```
Comment (AlanWong-MS) [inline src/Microsoft.OData.Edm/GlobalSuppressions.cs]: typo: **Authorizaation**VocabularyModel.#.cctor() -> **Authorization**VocabularyModel.#.cctor()

Will it be non-trivial to correct the name?

Comment (xuzhg) [reply src/Microsoft.OData.Edm/GlobalSuppressions.cs]: Thanks.

File: src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          </Record>
+          <Record>
+            <PropertyValue Property="rel" String="describedby" />
+            <PropertyValue Property="href"
+              String="https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Authorization.V1.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="Authorizations" Type="Collection(Auth.Authorization)" AppliesTo="EntityContainer EntitySet Singleton NavigationProperty Action Function">
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml]: AppliesTo should now only be "EntityContainer".

File: src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        <Annotation Term="Core.Description" String="Lists the methods available to authorize access to the annotated resource" />
+      </Term>
+
+      <ComplexType Name="Authorization" Abstract="true">
+        <Annotation Term="Core.Description" String="Base type for all Authorization types" />
+        <Property Name="Description" Type="Edm.String">
+          <Annotation Term="Core.Description" String="Description of the authorization method" />
+        </Property>
+        <Property Name="Name" Type="Edm.String">
+          <Annotation Term="Core.Description" String="Name that can be used to reference the authorization flow."/>
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml]: Nit; I would probably put the name before the description

File: src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            <PropertyValue Property="href"
+              String="https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Authorization.V1.md" />
+          </Record>
+        </Collection>
+      </Annotation>
+
+      <Term Name="Authorizations" Type="Collection(Auth.Authorization)" AppliesTo="EntityContainer EntitySet Singleton NavigationProperty Action Function">
+        <Annotation Term="Core.Description" String="Lists the methods available to authorize access to the annotated resource" />
+      </Term>
+
```
Comment (mikepizzo) [inline src/Microsoft.OData.Edm/Vocabularies/AuthorizationVocabularies.xml]: Please also add:
~~~xml
<Term Name="SecuritySchemes" Type="Collection(Auth.SecurityScheme)" AppliesTo="EntityContainer"> 
  <Annotation Term="Core.Description" String="At least one of the specified security schemes are required to make a request against the service."/> 
</Term> 
~~~
For enumerating the security schemes that apply to the entire service. See https://issues.oasis-open.org/browse/ODATA-1145, #4.

**Comment by xuzhg** [comment]
@mikepizzo @robward-ms I updated the PR and made the VSTS build pass. see at: https://identitydivision.visualstudio.com/DefaultCollection/OData/_build/index?buildId=85838&_a=summary

If you have no other questions, I will merge it into master branch. Thanks.

**Comment by xuzhg** [comment]
@mikepizzo Build passed. So I merged it and please let me know any further question.

**Comment by mikepizzo** [comment]
Looks good; thanks Sam and Rob.

**Pull request merged**